tests/replication/tcp/test_handler.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2022 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from tests.replication._base import BaseMultiWorkerStreamTestCase
+from tests.unittest import HomeserverTestCase, override_config
+
+try:
+    import hiredis
+except ImportError:
+    hiredis = None  # type: ignore
+
+
+class ChannelsMainTestCase(HomeserverTestCase):
+    if not hiredis:
+        skip = "Requires hiredis"
+
+    @override_config({"redis": {"enabled": True}})
+    def test_subscribed_to_enough_redis_channels(self) -> None:
+        # The default main process is subscribed to USER_IP and all RDATA channels.
+        self.assertCountEqual(
+            self.hs.get_replication_command_handler()._channels_to_subscribe_to,
+            ["USER_IP"],
+        )
```

Comment:
Can we add a test that checks we do in fact connect to these channels? Presumably we can look at what's in `FakeRedisPubSubServer`?

Reply:
Good point.

I've added a new base test case class that enables Redis (because even if you're not using it with the main process for anything, you must enable it in order for the Fake Redis server to listen on a fake TCP port) and skips Postgres (since it turns out nothing subscribes to Redis when Postgres is not in use...).